### DISCUSSION TITLE
Problem with 'from'-Property and 'id' under layout routes

### DISCUSSION THREAD

**Discussion by nilshartmann** [TanStack/router#920] [Q&A] [answered]
Hi,

when using the `Link` component (or `navigate`) where the `from` property is set to the `id` of a route that is a children of a layout route, the `search` params are `never`.

```
navigate({
            from: myFromRoute.id
            search: (s) => ({ // s is never! })
});
```

When I use `myFromRoute.fullPath` instead of `myFromRoute.id` it seems to work, but I'm not sure, if that is correct, as it states [in the docs](https://tanstack.com/router/v1/docs/guide/navigation#tooptions-interface): "It's common to supply the route.id of the origin route you are rendering "

Anyone has an idea?

Thanks,
Nils

**Comment by schiller-manuel** [comment, marked as answer]
You are right, `.fullPath` is the correct solution. We will update the docs.

**Reply by nilshartmann** [reply]
Thanks a lot for your reply!

Just to be sure: for the `to` property i would use `to` from the route (`<Link to={myRoute.to}>...</Link>`)?

**Reply by schiller-manuel** [reply]
yes

**Comment by nilshartmann** [comment]
Hi @schiller-manuel ,

I can't find the `id` and `fullPath` recommandations in the docs anymore. Is their usage still recommended? 

Thanks,
Nils

**Reply by schiller-manuel** [reply]
It's still there, kind of :)

https://tanstack.com/router/v1/docs/framework/react/guide/navigation#tooptions-interface

>   // `from` is an optional route ID or path. If it is not supplied, only absolute paths will be auto-completed and type-safe. It's common to supply the route.fullPath of the origin route you are rendering from for convenience. If you don't know the origin route, leave this empty and work with absolute paths or unsafe relative paths.

However, we saw lots of TS errors originating from circular dependencies, especially when other routes where imported for the `to` property.
So I'd say: within a route, you can use `Route.fullPath` for the `from` property, but I advise against importing other routes.
Instead, use the string literals (`/path/to/$fooId`).

**Reply by nilshartmann** [reply]
Thanks a ton!

> It's still there, kind of :)

I tried the search, and that still includes references to the original link that does not exist anymore :-)

> but I advise against importing other routes.

That would have been my next question. As with file based routing every route is named exported as `Route` it became quite hard to import the correct one. But if importing is not considered a good practice aynmore, it doesn't matter anyway.
